### DISCUSSION TITLE
Rustdesk with libp2p-rust

### DISCUSSION THREAD

**Discussion by dvasanth** [rustdesk/rustdesk#9438] [General]
Libp2p-rust uses bootstrap servers to join the IPFS DHT, where they can discover and connect to other peers in a decentralized network. It supports hole punching and help connect to nodes behind NAT.
I'm considering porting RustDesk to use libp2p streams, making it a fully decentralized app that would work without public servers. Has anyone explored this idea before? Would it be a valuable feature to add?

**Comment by dvasanth** [comment]
Added support over libp2p in hopdesk https://gitlab.com/d.vasant/hoptodesk/-/merge_requests/1/diffs
Since the code base is similar, you can backport it if anyone needs it. The above change uses IPFS bootstrap network to establish p2p (hole punched) connection  or relayed connection if not possible. It can make the p2p connection establishment process simpler without any servers in a decentralized way.

**Comment by aabittner** [comment]
in the wake of constant ddos attacks on public rendezvous server, i call for a DHT option for rustdesk to find its fellow peers and nodes, i am offering a bounty to make this happen at last, we need for example using the mainline DHT network (bittorrent folks, but many other apps use it too for bootstrapping nodes and services).

please guys, can anyone implement a robust and proper way and skip the rendezvous server and all. we need modern solutions and not overloaded servers and server backend counterparts for rustdesk.

anyone?  see ghostly application for example here on github as well, which makes perfect use of the mainline DHT network

https://github.com/MiguelMedeiros/ghostly

thanks for consideration.
